datafusion/core/tests/sql/timestamp.rs

```diff
@@ -1742,5 +1741,55 @@ async fn test_ts_dt_binary_ops() -> Result<()> {
 
     assert_batches_eq!(expected, &results);
 
+    //test cast path timestamp date using literals
+    let sql = "select '2000-01-01'::timestamp >= '2000-01-01'::date";
```

Comment:
You might be able to add these tests to sqllogictest if you wanted -- https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sqllogictests/test_files/timestamps.slt 

Not required but I find the sqllogictest framework much easier to work with

Reply:
Thanks @alamb I will create a separate ticket to move all timestamp tests from timestamp.rs to sqllogictest, all that can be moved of course

Reply:
Thank you very much -- that would be great

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/5165